### ISSUE TITLE
Multi-GPU training

### ISSUE THREAD

**Issue by hegc** [k2-fsa/snowfall#76]
Hi, these examples are excellent, and can we training on multi-GPUs with K2/lhotse? Just like the DDP in pytorch?

**Comment by danpovey** [comment]
I don't believe we have examples of multi-GPU training yet, but AFAIK
standard PyTorch mechanisms for multi-GPU training should work.
If you try, let us know, and make a PR about it!

**Comment by csukuangfj** [comment]
There is a WIP pull-request about multi-GPU training: https://github.com/k2-fsa/snowfall/pull/71

**Comment by hegc** [comment]
Thanks, I'll try it. @danpovey @csukuangfj

**Comment by pzelasko** [comment]
BTW I intend to finish that PR so that we can switch between single-GPU and multi-GPU training when I find some spare time. I will also need to make sure that it does the right thing when aggregating things like validation loss between the GPUs (I saw both processes return slightly different values) and storing/loading checkpoints. If you have more time and can pick it up before I do, you're welcome to do it :)

**Comment by danpovey** [comment]
Fantastic!!
